roachprod: ensure world-readable local files in get

tbg opened the pull request:
CRDB intentionally runs under a restrictive umask, which was likely to
trip up poorly configured CI environments in which files are created
under a different user than they are later read. Ideally such CI
environments would be fixed up, but there is an urgent need to work
around the problem at this very moment.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/44876)
<!-- Reviewable:end -->

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> modulo possibly missing return.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/44876#-:-M-WR9RzE0dENQc2rHLq:b7upww4)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @petermattis and @tbg)

___
*[pkg/cmd/roachprod/install/cluster_synced.go, line 1366 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/44876#-M-WQwH89eZeNfuIAfC2:-M-WQwH89eZeNfuIAfC3:b-chliph) ([raw file](https://github.com/cockroachdb/cockroach/blob/a68e93010756d120aec41cfdcb693367755c164c/pkg/cmd/roachprod/install/cluster_synced.go#L1366)):*
> ```Go
> 			err := c.scp(fmt.Sprintf("%s@%s:%s", c.user(c.Nodes[0]), c.host(c.Nodes[i]), src), dest)
> 			if err != nil {
> 				results <- result{i, err}
> ```

Do you need to `return` here?


<!-- Sent from Reviewable.io -->

tbg reviewed (commented):
bors r=petermattis

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/44876#-:-M-WS9No4Rjm0K5C0M6X:blkd80i)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @petermattis)

___
*[pkg/cmd/roachprod/install/cluster_synced.go, line 1366 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/44876#-M-WQwH89eZeNfuIAfC2:-M-WS5KTE9IMONVxFuaI:bf7hpui) ([raw file](https://github.com/cockroachdb/cockroach/blob/a68e93010756d120aec41cfdcb693367755c164c/pkg/cmd/roachprod/install/cluster_synced.go#L1366)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Do you need to `return` here?
</blockquote></details>

Definitely, thanks for spotting this. I rearranged things to avoid this awkward flow.


<!-- Sent from Reviewable.io -->

tbg committed:
roachprod: ensure world-readable local files in get

CRDB intentionally runs under a restrictive umask, which was likely to
trip up poorly configured CI environments in which files are created
under a different user than they are later read. Ideally such CI
environments would be fixed up, but there is an urgent need to work
around the problem at this very moment.

Release note: None

tbg commented:
bors r=petermattis
(now I actually pushed)
